test/SlowTests/Client/TimeSeries/Query/TimeSeriesQueryTests.cs

```diff
@@ -118,5 +163,861 @@ select out(u)
+                        .AddParameter("start2", baseline2)
+                        .AddParameter("end2", baseline2.AddDays(1));
+
+                    var result = query.ToList();
+
+                    Assert.Equal(2, result.Count);
+
+                    for (int i = 0; i < 2; i++)
+                    {
+                        var agg = result[i];
+
+                        var heartrate = agg.HeartRate;
+
+                        Assert.Equal(3, heartrate.Count);
+
+                        Assert.Equal(1, heartrate.Results.Length);
+
+                        var val = heartrate.Results[0];
+
+                        Assert.Equal(59, val.Min);
+                        Assert.Equal(79, val.Max);
+
+                        Assert.Equal(baseline.AddMinutes(60), val.From);
+                        Assert.Equal(baseline.AddMinutes(120), val.To);
+
+                        var bloodPressure = agg.BloodPressure;
+
+                        Assert.Equal(3, bloodPressure.Count);
+
+                        Assert.Equal(1, bloodPressure.Results.Length);
+
+                        val = bloodPressure.Results[0];
+
+                        Assert.Equal(159, val.Min);
+                        Assert.Equal(179, val.Max);
+
+                        double expectedAvg = (159 + 168 + 179) / 3.0;
+
+                        Assert.Equal(expectedAvg, val.Avg);
+
+                        Assert.Equal(baseline2.AddMinutes(60), val.From);
+                        Assert.Equal(baseline2.AddMinutes(120), val.To);
+
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public void CanQueryTimeSeriesAggregation_SelectSyntax_AliasNotation()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var baseline = DateTime.Today;
+
+                using (var session = store.OpenSession())
+                {
+                    for (int i = 1; i <= 3; i++)
+                    {
+                        var id = $"people/{i}";
+
+                        session.Store(new Person
+                        {
+                            Name = "Oren",
+                            Age = i * 30
+                        }, id);
+
+                        var tsf = session.TimeSeriesFor(id);
+
+                        tsf.Append("HeartRate", baseline.AddMinutes(61), "watches/fitbit", new[] { 59d });
+                        tsf.Append("HeartRate", baseline.AddMinutes(62), "watches/fitbit", new[] { 79d });
+                        tsf.Append("HeartRate", baseline.AddMinutes(63), "watches/fitbit", new[] { 69d });
+                    }
+
+                    session.SaveChanges();
+                }
+
+                new PeopleIndex().Execute(store);
+
+                WaitForIndexing(store);
+
+                using (var session = store.OpenSession())
+                {
+                    var query = session.Advanced.RawQuery<RawQueryResult>(@"
+from People as p
+where p.Age > 49
+select timeseries(
```

Comment:
Needs to think how we can do this on _JS_ calls as well.

Reply:
will be handled separately in https://issues.hibernatingrhinos.com/issue/RavenDB-13082